test/integration/targets/aws_acm/meta/main.yml

```diff
@@ -0,0 +1,3 @@
+dependencies:
+  - setup_openssl
```

Comment:
This doesn't seem to work very well in the ansible-test container:

```
TASK [setup_openssl : Install pyOpenSSL] ***************************************
task path: /root/.ansible/test/tmp/aws_acm-t_0b90zn-ÅÑŚÌβŁÈ/test/integration/targets/setup_openssl/tasks/main.yml:6
<testhost> ESTABLISH LOCAL CONNECTION FOR USER: root
<testhost> EXEC /bin/sh -c 'echo ~root && sleep 0'
<testhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1565867400.6824777-102379632105809 `" && echo ansible-tmp-1565867400.6824777-102379632105809="` echo /root/.ansible/tmp/ansible-tmp-1565867400.6824777-102379632105809 `" ) && sleep 0'
Using module file /root/ansible/lib/ansible/modules/packaging/os/apt.py
<testhost> PUT /root/.ansible/tmp/ansible-local-132q9lvaufd/tmpdh2r6nem TO /root/.ansible/tmp/ansible-tmp-1565867400.6824777-102379632105809/AnsiballZ_apt.py
<testhost> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1565867400.6824777-102379632105809/ /root/.ansible/tmp/ansible-tmp-1565867400.6824777-102379632105809/AnsiballZ_apt.py && sleep 0'
<testhost> EXEC /bin/sh -c 'ANSIBLE_DEBUG_BOTOCORE_LOGS=True /tmp/python-qme69j5a-ansible/python /root/.ansible/tmp/ansible-tmp-1565867400.6824777-102379632105809/AnsiballZ_apt.py && sleep 0'
<testhost> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-tmp-1565867400.6824777-102379632105809/ > /dev/null 2>&1 && sleep 0'
 [WARNING]: Updating cache and auto-installing missing dependency: python3-apt

The full traceback is:
  File "/tmp/ansible_apt_payload_n70gyc60/__main__.py", line 1038, in main
    import apt
  File "/usr/lib/python3/dist-packages/apt/__init__.py", line 23, in <module>
    import apt_pkg

fatal: [testhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "allow_unauthenticated": false,
            "autoclean": false,
            "autoremove": false,
            "cache_valid_time": 0,
            "deb": null,
            "default_release": null,
            "dpkg_options": "force-confdef,force-confold",
            "force": false,
            "force_apt_get": false,
            "install_recommends": null,
            "name": "python3-openssl",
            "only_upgrade": false,
            "package": [
                "python3-openssl"
            ],
            "policy_rc_d": null,
            "purge": false,
            "state": "present",
            "update_cache": null,
            "upgrade": null
        }
    },
    "msg": "Could not import python modules: apt, apt_pkg. Please install python3-apt package."
}

PLAY RECAP *********************************************************************
testhost                   : ok=2    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
```

honestly never used this role

The other problem with it is I later get another warning:
```
ERROR: Test target "aws_acm" has changed the test environment!
If these changes are necessary, they must be reverted before the test finishes.
```
This should probably all be done in a virtual environment (perhaps rather than having setup_openssl as a dependency you could include the role after setting up a virtualenv in tasks/main.yml)

Reply:
Where did you see that error message?

Why does it not work? This is exactly how the crypto modules I am using do their tests.

How do I set up a virtual environment within a playbook, and run the subsequent tasks _within_that environment?

Reply:
I'm running `ansible-test integration --docker -vvv aws_acm` and that's what happens.
 
See e.g. ecs_cluster's test suite for how to run a test suite inside a virtualenv

Reply:
Ok, I just pushed a new commit to do that. When I run that command I get an authorization error.

> Couldn't obtain certificates: An error occurred (UnrecognizedClientException) when calling  the ListCertificates operation: The security token included in the request is invalid", resource_actions": ["acm:ListCertificates"].

I put some AWS creds in `cloud-config-aws.ini`. Was there something else I have to do?

Reply:
Just ensure security_token is set to "" if one isn't required.

Reply:
I already did that.

Reply:
```
# This is the configuration template for ansible-test AWS integration tests.
#
# You do not need this template if you are:
#
# 1) Running integration tests without using ansible-test.
# 2) Using the automatically provisioned AWS credentials in ansible-test.
#
# If you do not want to use the automatically provisioned temporary AWS credentials,
# fill in the @VAR placeholders below and save this file without the .template extension.
# This will cause ansible-test to use the given configuration instead of temporary credentials.
#
# NOTE: Automatic provisioning of AWS credentials requires an ansible-core-ci API key.

[default]
aws_access_key: ABCDE
aws_secret_key: 123456
security_token: ""
aws_region: ap-southeast-2
# aws_cleanup controls whether the environment is cleaned up after tests have completed
# This only applies to tests that have a cleanup stage
# Defaults to true when using this template
# aws_cleanup: true
# aliases for backwards compatibility with older integration test playbooks
ec2_access_key: {{ aws_access_key }}
ec2_secret_key: {{ aws_secret_key }}
ec2_region: {{ aws_region }}
```

Reply:
Sorry, I wasn't on my test machine. Apparently it's `security_token:` - ie. nothing after the colon

Reply:
Yes that `security_token` bit fixed it.